base/linalg/triangular.jl

```diff
@@ -995,15 +1048,97 @@ function logm{T<:Union{Float64,Complex{Float64}}}(A0::UpperTriangular{T})
          1.276404810806775e-001,
          2.060962623452836e-001,
          2.879093714241194e-001]
+
+    n = chksquare(A0)
+
+    A, m, s = invsquaring(A0,theta)
+
+    # Compute accurate diagonal of T - I
+    sqrt_diag!(A0, A, s)
+
+    # Compute the Gauss-Legendre quadrature formula
+    R = zeros(Float64, m, m)
+    for i = 1:m - 1
+        R[i,i+1] = i / sqrt((2 * i)^2 - 1)
+        R[i+1,i] = R[i,i+1]
+    end
+    x,V = eig(R)
+    w = Array(Float64, m)
+    for i = 1:m
+        x[i] = (x[i] + 1) / 2
+        w[i] = V[1,i]^2
+    end
+
+    # Compute the Padé approximation
+    Y = zeros(T, n, n)
+    for k = 1:m
+        Y = Y + w[k] * (A / (x[k] * A + I))
```

Comment:
Again a lot of matrix allocations here; would be better to replace with in-place updates of `Y` etc.